waspc/src/Wasp/Generator/NpmInstall.hs

```diff
@@ -0,0 +1,106 @@
+module Wasp.Generator.NpmInstall
```

Comment:
My general feeling for this file is that it is still somewhat hard to understand exactly what is happening -> there is a lot of functions with similar names, some assumptions, and similar. It is pretty complex.

How can we make it even more readable / understandable?

For the start maybe we could write one comment depicting the whole picture?

Reply:
I've done some more renaming and commenting.